packages/reason-css-parser/ppx/Generate.re

```diff
@@ -636,5 +658,175 @@ module Make = (Ast_builder: Ppxlib.Ast_builder.S) => {
     | Combinator(kind, values) => combinator_op(kind, values)
     | Function_call(name, value) => function_call(name, value)
     };
-  };
+  };  
+
+let rec (create_renderer : value => Parsetree.expression) = (value) => {
+	let apply_modifier = (modifier, value) => switch(modifier){
+			| One => value
+			| Optional => pexp_fun(Nolabel, None, [%pat? arg], pexp_apply([%expr Option.bind], [(Nolabel, [%expr arg]), (Nolabel, value)]))
+			| Repeat(_)
+			| Repeat_by_comma(_, _)
+			| Zero_or_more
+			| One_or_more => pexp_apply([%expr List.map], [(Nolabel, value)])
+			| At_least_one => value
+		};
+
+	let rec terminal_op = (kind, _multiplier) => {
+			switch (kind) {
+			| Delim(name) =>
+				let variant_name = value_of_delimiter(name);
+				let lhs = ppat_variant(variant_name, None);
+				let rhs = name |> value_of_delimiter |> pconst_string |> pexp_constant;
+				let cases = case(~lhs, ~rhs, ~guard=None);
+				pexp_function([cases]);
+
+			| Keyword(name) =>
+				let variant_name = variant_name(value);
+				let lhs = ppat_variant(variant_name, None);
+				let rhs = name |> pconst_string |> pexp_constant;
+				let cases = case(~lhs, ~rhs, ~guard=None);
+				pexp_function([cases]);
+			| Property_type(name) 
+			| Data_type(name) =>
+				let formatted_name = value_name_of_css(name);
+				let name = Lident("render_" ++ formatted_name);
+				pexp_apply(pexp_ident @@ txt(name), []);
+			};
+		}
+
+	and terminal_xor_op = (kind, value) => {
+			switch (kind) {
+			| Delim(name) => 
+				{
+				let formatted_name = variant_name(value);
+				(ppat_variant(formatted_name, None), name |> value_of_delimiter |> pconst_string |> pexp_constant);
+				}
+
+			| Keyword(name) =>
+				let formatted_name = variant_name(value);
+				(ppat_variant(formatted_name, None), name |> pconst_string |> pexp_constant);
+			| Property_type(name) 
+			| Data_type(name) =>
+				let formatted_name = value_name_of_css(name);
+				let name = Lident("render_" ++ formatted_name);
+				(ppat_variant(variant_name(value), Some([%pat? arg])), pexp_apply(pexp_ident @@ txt(name), [(Nolabel, pexp_ident(txt(Lident("arg"))))]))
+			};
+	}
+
+	and variant_of_values = (kind, values) => {
+		let string_of_kind = fun
+		| Xor => "Xor"
+		| Static => "Static"
+		| Or => "Or"
+		| And => "And"
+		
+		let args = List.mapi((i, _) => Format.sprintf("arg_%d", i) |> txt |> ppat_var, values); 
```

Comment:
instead of mapping thought args, might be worth trying the approach I mention and use the name for a render_* that already exists (or to try to create a render)